sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication.py

```diff
@@ -46,6 +49,11 @@ class ARMChallengeAuthenticationPolicy(BearerTokenCredentialPolicy):
     :param str scopes: required authentication scopes
     """
 
+    def __init__(self, credential: "TokenCredential", *scopes: str, **kwargs: Any) -> None:
+        # ARM supports Continuous Access Evaluation (CAE). This policy will enable it by default.
+        kwargs.setdefault("enable_cae", True)
```

Comment:
+1, what is the current behavior?

Reply:
Also CAE is documented as defaulting to false in credentials, should it be consistent between policy and credential?

Reply:
The current behavior is that Identity currently _always_ requests CAE-enabled tokens when users use certain types of credentials (e.g. User Credentials like `InteractiveBrowserCredential`, sync service principal credentials like `ClientSecretCredential`). This "always-on" mode is kinda a broken state that we are now moving away from. 

ARM actually supports CAE on the service side and SDK side (by handling CAE challenge claims), so defaulting `enable_cae` to true maintains this behavior of CAE tokens being requested for these credentials. 

I suppose one change that potentially could be noted is that if some mixed combination of azure-core, azure-mgmt-core, and azure-identity are used, then it's possible for non-CAE tokens to be requested, but I 

> Also CAE is documented as defaulting to false in credentials, should it be consistent between policy and credential?

The base policies still default to false. The ARM policy is defaulted to True intentionally so that all ARM SDKs can continue to get CAE tokens.

Reply:
Currently how mgmt SDKs opt in CAE?

Reply:
I don't think using different default for core and mgmt-core is a good idea.

Reply:
> Currently how mgmt SDKs opt in CAE?

Currently, they are opted-in when a user uses one of the credentials where we always enable CAE. Remember that the current CAE-support for Identity is to always pass in `CP1` client capabilities in the token request regardless if the resource API or SDK supported CAE or not.

> I don't think using different default for core and mgmt-core is a good idea.

If we want all the mgmt SDKs to have this enabled, the options are:

1) To have each individual instantiation of `ARMChallengeAuthenticationPolicy` across each mgmt SDK pass in `enable_cae=True`. This would then need a release for each ARM SDK package so that the change can be reflected.
2) Set `enable_cae=True` in the `ARMChallengeAuthenticationPolicy` itself. Now, as long as users have the latest version of `azure-mgmt-core`, they will have CAE enabled tokens for credentials that support it across any mgmt sdk. Seems to me that this option is much more tenable.

Reply:
In other words, mgmt SDKs did not ask us to enable CAE explicitly.

They are enabled because we turned it on by default, right?

Reply:
Right, around two years ago. CAE support was added for just user credentials in Identity (https://github.com/Azure/azure-sdk-for-python/pull/16323) where user-credentials always requested CAE tokens. Then the `ARMChallengeAuthenticationPolicy` [was soon created](https://github.com/Azure/azure-sdk-for-python/pull/19365) afterwards to handle any CAE claims challenges for those using user credentials. No explicit enablement from the side of the mgmt sdk. 

Since the paradigm is now shifting towards enabling CAE per-request with CAE disabled by default, this change allows credentials to still get CAE-enabled tokens when interacting with ARM SDKs.

Reply:
Splitting mgmt-core changes into a separate PR for future review/discussion.